### PR TITLE
Cleaner2

### LINKED ISSUE #166: When the text contained in an mjml tag contains the character '<' an exception is triggered
When the text contained in an mjml tag contains the character '<' an exception is triggered.

Sample: 
```HTML
<mjml>
    <mj-body>
        <mj-text>test: &, <, >, ', \"</mj-text>
    </mj-body>
</mjml>
```

You should add the conversion of the '<' to '& lt;' as for the '&'

I added this condition in XmlFixer, it seems to work:
 
```C#
else if (span.Length >= 2 && span[0] == '<' && span[1] != '/' // not closing tag
    && !((span.Length >= 4 && span[1] == 'm' && span[2] == 'j') // ignore start tag
        || (span.Length >= 3 && span[2] == 'm' && span[3] == 'j'))) // ignore end tag
{
    if (!IsEntity(span))
    {
        sb.Append("&lt;");
    }
    else
    {
        sb.Append(first);
    }
}
```

@SebastianStehle what do you think?

### PR DESCRIPTION
**Pull request by SebastianStehle**
Better HTML fixer. This uses the html tokenizer from AngleSharp. The tokenizer is internal, so a made private fork for that.

For backwards compatibility I have introduced a new options.

Closes #166 

Edit: It uses reflection now with IL emit, so that we do not have to maintain our fork.

### CODE REVIEW

**Comment by LiamRiddell** [comment]
I'll review tomorrow 👍🏻

File: Tests/XmlFixerTests.cs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        public void Should_convert_self_closing_tag3()
+        {
+            var input = @"
+<div>
+    <br> </br>
+</div>";
+
+            var expected = @"
+<div>
+    <br> </br>
```
Comment (LiamRiddell) [inline Tests/XmlFixerTests.cs:51]: In this test case, we should be expecting `<br/>` (no whitespace) instead of `<br> </br>`?

**Review by LiamRiddell — CHANGES_REQUESTED** [review]
Overall, it looks really clean. The decision to go with IL Emit makes sense instead of reflection.

I've left a few suggestions (mostly for comments) and comments.

File: Tests/XmlFixerTests.cs (hunk 1 above)
Comment (SebastianStehle) [reply Tests/XmlFixerTests.cs:51]: No, it should keep everything intact.

File: Mjml.Net/XmlFixer.cs [hunk 2]
```diff
@@ -1,10 +1,195 @@
...
+                            // <br> </br>
+                            else if (IsMatchingToken(startTag, next2) && next1.Type == HtmlTokenType.Character)
+                            {
+                                // If there is some text between the tokens we have to fix the name.
+                                sb.Append('>');
+                                WriteToken(sb, next1, htmlReader);
+                                sb.Append(CultureInfo.InvariantCulture, $"</{startTag.Name}>");
+                            }
+                            // <br>...
+                            else
```
[the remark this answers is not in the document]
Comment (SebastianStehle) [reply Mjml.Net/XmlFixer.cs:124]: There is a case for that. The character token is actually for multiple characters afaik.

[the remark this answers is not in the document]
Comment (SebastianStehle) [reply Mjml.Net/XmlFixer.cs:124]: Even if not, I would not make it even more complex.

**Review by LiamRiddell — APPROVED** [review]
Awesome, great work @SebastianStehle

**Pull request merged**